app/assets/javascripts/components/markdown-editor.js

```diff
@@ -1,2 +1,70 @@
 //= require vendor/@webcomponents/webcomponentsjs/webcomponents-loader.js
 //= require vendor/@github/markdown-toolbar-element/dist/index.umd.js
+//= require vendor/marked/lib/marked.js
+
+function MarkdownEditor ($module) {
+  this.$module = $module
+  this.$input = $module.querySelector('textarea')
+  this.$preview = $module.querySelector('.js-preview-body .govuk-textarea')
+  this.$previewButton = $module.querySelector('.js-preview-button')
+  this.$editButton = $module.querySelector('.js-edit-button')
+  this.$editorInput = $module.querySelector('.js-editor-input')
+  this.$previewBody = $module.querySelector('.js-preview-body')
+}
+
+MarkdownEditor.prototype.init = function () {
+  var $module = this.$module
+
+  // Save bounded functions to use when removing event listeners during teardown
+  $module.boundPreviewButtonClick = this.handlePreviewButton.bind(this)
+  $module.boundEditButtonClick = this.handleEditButton.bind(this)
+
+  // Handle events
+  this.$previewButton.addEventListener('click', $module.boundPreviewButtonClick)
+  this.$editButton.addEventListener('click', $module.boundEditButtonClick)
+}
+
+MarkdownEditor.prototype.handlePreviewButton = function (event) {
+  event.preventDefault()
+
+  var $preview = this.$preview
+  var text = this.$input.value
+
+  // Mirror textarea's height
+  $preview.style.height = this.$input.offsetHeight + 'px'
+
+  // Render markdown
+  window.marked(text, function (err, content) {
+    if (err) throw err
```

Comment:
What kind of errors are we anticipating here - should we be handling them?

Reply:
marked can return parsing errors and if they happen I'm thinking we don't want them to fail silently.

Reply:
Totally agree we shouldn't forget about the error, but how will the user know they've done something wrong?

Reply:
They're evaluating the preview coming from content, but they may not realise that indeed. I added 'Error previewing content' as a placeholder if marked throws an error.